### ISSUE TITLE
[Issue]: build fails due to cppheaderparser package missing but it is installed

### ISSUE THREAD

**Issue by jdgh000** [ROCm/clr#107]
### Problem Description

```
-- Found Python3: /bin/python3.12 (found version "3.12.6") found components: Interpreter
Traceback (most recent call last):
  File "<string>", line 1, in <module>
ModuleNotFoundError: No module named 'CppHeaderParser'
CMake Error at hipamd/src/CMakeLists.txt:246 (message):
      The "CppHeaderParser" Python3 package is not installed.     Please install it using the following command: "pip3 install CppHeaderParser".

 pip list | grep -i CppHeaderParser
CppHeaderParser          2.7.4

rocm6.2.3, centos 9
```

### Operating System

centos

### CPU

amd

### GPU

mi250

### ROCm Version

ROCm 6.2.3

**Comment by cjatin** [comment]
Can you share the output of `pip --version`

**Comment by mangupta** [comment]
Also could you try running ```pip3 list | grep -i CppHeaderParser```?

**Comment by jdgh000** [comment]
pip 24.2

**Comment by cjatin** [comment]
> pip 24.2

The full output, something like:

`pip 24.0 from /usr/lib/python3/dist-packages/pip (python 3.12)`

**Comment by jdgh000** [comment]
Name: pip
Version: 24.3.1
Summary: The PyPA recommended tool for installing Python packages.
Home-page: https://pip.pypa.io/
Author:
Author-email: The pip developers <distutils-sig@python.org>
License: MIT
Location: /usr/local/lib/python3.9/site-packages

**Comment by jdgh000** [comment]
I tried several diferent pip version all fail, what are you trying to look here from pip version?

**Comment by jdgh000** [comment]
I found it may be looking at different version of python softlink under /bin as I see several version. 
But that may not be good idea "which python" points to different python version (3.12). It may be that your code is looking at latest version. Not sure.
But more stable approach would be using the one that is searched and found by "which python3" command which points to 3.9 in my node.

**Comment by darren-amd** [comment]
Hi @jdgh000,

This appears to be a similar issue as https://github.com/ROCm/roctracer/issues/108. Could you please confirm if the fix highlighted here: https://github.com/ROCm/roctracer/issues/108#issuecomment-2492576989 resolves your issue? Thanks!

**Comment by jdgh000** [comment]
-DPython3_EXECUTABLE= appears to work

**Comment by jdgh000** [comment]
> Hi @jdgh000,
> 
> This appears to be a similar issue as [ROCm/roctracer#108](https://github.com/ROCm/roctracer/issues/108). Could you please confirm if the fix highlighted here: [ROCm/roctracer#108 (comment)](https://github.com/ROCm/roctracer/issues/108#issuecomment-2492576989) resolves your issue? Thanks!

yes, i opened it, both roctracer and clr is fixed by same -D parameter specifying python path: /usr/bin/python3
